server/src/main/java/org/elasticsearch/index/IndexVersions.java

```diff
@@ -243,8 +236,6 @@ static NavigableMap<Integer, IndexVersion> getAllVersionIds(Class<?> cls) {
         return Collections.unmodifiableNavigableMap(builder);
     }
 
-    @UpdateForV9(owner = UpdateForV9.Owner.CORE_INFRA)
-    // We can simplify this once we've removed all references to index versions earlier than MINIMUM_COMPATIBLE
```

Comment:
We shouldn't remove old index versions here - this is something the search team need to take a look at